weblab/static/js/db.js

```diff
@@ -499,42 +499,41 @@ function parseLocation ()
       if (items[0] == "public")
       {
         $('#showPublicExpts').addClass("selected");
-        ret.publicOnly = "1";
+        ret.subset = "public";
       }
-      else if (items[0].substr(0,4) == "mine")
+      else if (items[0] == "mine")
       {
         $('#showMyExpts').addClass("selected");
         $('.showMyButton').show();
-        ret.mineOnly = "1";
-        if (items[0].indexOf("-m") != -1)
-        {
-          ret.includeModeratedModels = "0";
-          console.log('Show model');
+        ret.subset = "mine";
+
+        $('#showMyExptsModels').text("Hide moderated models");
+        $('#showMyExptsProtocols').text("Hide moderated protocols");
+
+        var query = location.search.substr(1);
+        var result = {};
+        query.split("&").forEach(function(part) {
+          var item = part.split("=");
+          if (item[0] == 'moderated-models' && item[1] == 'false') {
+            ret["moderated-models"] = "false";
           $('#showMyExptsModels').text("Show moderated models");
-        }
-        else
-        {
-          $('#showMyExptsModels').text("Hide moderated models");
-        }
-        if (items[0].indexOf("-p") != -1)
-        {
-          ret.includeModeratedProtocols = "0";
-          console.log('Show proto');
-          $('#showMyExptsProtocols').text("Show moderated protocols");
-        }
-        else
-        {
-          $('#showMyExptsProtocols').text("Hide moderated protocols");
-        }
+          }
+          if (item[0] == 'moderated-protocols' && item[1] == 'false') {
+            ret["moderated-protocols"] = "false";
+            $('#showMyExptsProtocols').text("Show moderated protocols");
+          }
+          result[item[0]] = decodeURIComponent(item[1]);
+        });
       }
-      else if (items[0] == "all")
+      else if (items[0] == "moderated")
       {
-        $('#showAllExpts').addClass("selected");
-        ret.showAll = "1";
+        $('#showModeratedExpts').addClass("selected");
+        ret.subset = "moderated";
       }
       else
```

Comment:
This shouldn't be the default case; instead it should be `if (items[0] == "all")`